communication/nats/receiver.go

```diff
@@ -66,12 +70,28 @@ func (receiver *receiverNATS) Receive(consumer communication.MessageConsumer) er
 		}
 	}
 
-	_, err := receiver.connection.Subscribe(messageTopic, messageHandler)
+	receiver.mu.Lock()
+	defer receiver.mu.Unlock()
+
+	subscription, err := receiver.connection.Subscribe(messageTopic, messageHandler)
 	if err != nil {
 		err = fmt.Errorf("failed subscribe message '%s'. %s", messageTopic, err)
 		return err
 	}
+	receiver.subs[messageTopic] = subscription
+	return nil
+}
+
+func (receiver *receiverNATS) Unsubscribe() error {
+	receiver.mu.Lock()
+	defer receiver.mu.Unlock()
 
+	for topic, s := range receiver.subs {
+		if err := s.Unsubscribe(); err != nil {
```

Comment:
If one of subscribers returns error, do we really want to skip unsubscribing of others? Maybe just log error and continue? Or maybe unsubscribe doesn't need to return error at all?

Reply:
Changed it to just log errors.